clients/drcachesim/tracer/instru_online.cpp

```diff
@@ -178,6 +178,12 @@ online_instru_t::append_thread_header(byte *buf_ptr, thread_id_t tid,
     new_buf += append_marker(new_buf, TRACE_MARKER_TYPE_FILETYPE, file_type);
     new_buf += append_marker(new_buf, TRACE_MARKER_TYPE_CACHE_LINE_SIZE,
                              proc_get_cache_line_size());
+#if defined(AARCH64)
+    if (proc_has_feature(FEATURE_SVE)) {
+        new_buf += append_marker(new_buf, TRACE_MARKER_TYPE_DYNAMIC_VECTOR_LENGTH,
```

Comment:
Would it be simpler to just add the marker on other archs too? Helps keep the traces consistent IMHO. @derekbruening

Reply:
Just a TRACE_MARKER_TYPE_VECTOR_LENGTH, and extend proc_get_vector_length_bytes() to other arches so on x86 it would tell you whether AVX-512 is there or not?  I could see that.  It would eliminate ifdefs in the test output above.  All the marker processing code is already not inside ifdefs which is good.  I don't feel strongly about it though: could live with it being aarch64 only.

Reply:
> could live with it being aarch64 only
It is only used on AArch64 at the moment but RISC-V also has a vector length that can be changed at runtime so it might be useful for that too in the future.

The first version of this patch actually did call the maker `TRACE_MARKER_TYPE_VECTOR_LENGTH` and tried to emit it on all architectures but I changed it to the way it is now when I started testing it on x86 and realised `proc_get_vector_length_bytes()` is aarch64-only. I didn't want to have to define what `proc_get_vector_length_bytes()` means for every architecture. It's easier to start with a narrow, well defined scope and broaden it later than it is to start with a broad scope and narrow/change it later if your broad definition turns out to be wrong.

Reply:
The thing that is hard to change later is the name of the marker, so if we think we do want to generalize proc_get_vector_length_bytes() it seems best to remove the _DYNAMIC_ from the name even if the implementation starts out aarch64-only (that iniitial limitation would be noted in the marker docs).

Reply:
I have changed it to `TRACE_MARKER_TYPE_VECTOR_LENGTH`